src/main/java/dk/cngroup/lentils/service/ImageService.java

```diff
@@ -89,13 +90,33 @@ public Resource loadAsResource(final String filename) {
         }
     }
 
-    public String getFilePathToSave(final MultipartFile file) {
+    private String getFileName(final MultipartFile file) {
         if (file.isEmpty()) {
             throw new FileNotFoundException();
         }
-        String fileName = file.getOriginalFilename();
-        FileUtils.createDirectory(getDirectory());
-        return getDirectory() + fileName;
+        return addHashToFilename(file.getOriginalFilename());
+    }
+
+    public String addHashToFilename(final String filename) {
+        int i = filename.lastIndexOf('.');
+        if (i < 0) {
+           return StringUtils.cleanPath(filename + generateHash());
+        } else {
+            return StringUtils.cleanPath(filename.substring(0, i)
+                    + "-" + new SimpleDateFormat("yyyy-MM-dd").format(new Date())
+                    + "-" + RandomStringUtils.randomAlphanumeric(HASH_LENGTH)
+                    + filename.substring(i));
+        }
+    }
+
+    private String generateHash() {
+        return "-" + new SimpleDateFormat("yyyy-MM-dd").format(new Date())
```

Comment:
Optional: Pouziti String.format a nebo StringuBuilder by bylo asi vhodne, protoze tam jsou uz 3 konkatenace. S pouzitim String.format by to mohlo byt taky prehlednejsi.